terraform/modules/s3/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "juju_application" "minio" {
+  # Coordinator requires s3
+  name  = var.minio_app
+  model = var.model_name
+  trust = true
+
+  charm {
+    name    = var.minio_app
+    channel = var.channel
+  }
+  units = 1
+
+  config = {
+    access-key = var.minio_user
+    secret-key = var.minio_password
+  }
+}
+
+resource "null_resource" "s3management" {
+  triggers = {
+    # model_name = var.model_name
+    always_run = timestamp()
+  }
+  provisioner "local-exec" {
+    environment = {
+      MINIO_USER     = var.minio_user
+      MINIO_PASSWORD = var.minio_password
+    }
+    command = <<-EOT
+      bash "${path.module}/scripts/s3management.sh" \
```

Comment:
This might be tricky to handle if depends on Minio application to be active :|

Reply:
I have separated the concerns here:
- main.tf execute
- s3management.sh has the logic.

In that last file we [wait for `minio` for be active](https://github.com/canonical/observability/pull/227/files#diff-0511ca4a126b5b68d1a737eff623410ff4459e4025fbc221a4607854d6c79a63R52).

Right now it works, but if you think there is a best way to do this, please let us know!